core/src/apps/debug/__init__.py

```diff
@@ -33,15 +33,19 @@
 
         while True:
             msg = await debuglink_decision_chan.take()
-
             if msg.yes_no is not None:
                 await confirm_chan.put(
                     confirm.CONFIRMED if msg.yes_no else confirm.CANCELLED
                 )
-            if msg.up_down is not None:
-                await swipe_chan.put(
-                    swipe.SWIPE_DOWN if msg.up_down else swipe.SWIPE_UP
-                )
+            if msg.swipe is not None:
+                if msg.swipe == DebugSwipeDirection.UP:
+                    await swipe_chan.put(swipe.SWIPE_UP)
+                elif msg.swipe == DebugSwipeDirection.DOWN:
+                    await swipe_chan.put(swipe.SWIPE_DOWN)
+                elif msg.swipe == DebugSwipeDirection.LEFT:
+                    await swipe_chan.put(swipe.SWIPE_LEFT)
+                elif msg.swipe == DebugSwipeDirection.RIGHT:
+                    await swipe_chan.put(swipe.SWIPE_RIGHT)
```

Comment:
maybe add
```
else
    raise ValueError
```

Reply:
I didn't want to do this, because raising an exception here will permanently terminate the debuglink_decision_dispatcher task. This would mean that all future DebugLinkDecision messages would be ignored. I assume that if this happens in the CI, then the CI would keep running on and on until it times out. In that case I think it's better to exit the emulator right away.

I also assume that we are checking the incoming messages as soon as we get them, so the emulator will respond with an error if the enum is out of range. So this else branch would only be reached if we extend the enum, but forget to handle the new values here.

How about logging an error: "Unrecognized swipe direction."?

Reply:
Let's keep it as it is then (without logging even).